test_deploy_heroku.py

```diff
@@ -142,34 +145,46 @@ def test_deploy_heroku_command(self):
         result = deploy_heroku_command("sha", "bar", "foobar", rollback=True)
         self.assertEqual(result, expected_result)
 
-    @patch("deploy_heroku.get_latest_heroku_release", return_value=MOCK_RELEASE)
-    @patch("deploy_heroku.get_response", return_value={"commit": MOCK_COMMIT_HASH})
-    @patch("deploy_heroku.trigger_release_retry", return_value=MOCK_RELEASE)
-    @patch("deploy_heroku.wait_for_release", return_value=None)
-    def test_main_redeploy(self, mock_wait_for_release, mock_trigger_release, mock_get_response, mock_get_release):
-        main(MOCK_HEROKU_APP, MOCK_HEROKU_API_TOKEN, MOCK_COMMIT_HASH, json.dumps({"ghd": {"type": "redeploy"}}))
-        mock_get_release.assert_called_with("fake-app", "fake-token")
-        mock_get_response.assert_called_with("fake-app", "fake-token", "slugs/227c9e81-a699-9a1bea3b8119")
-        mock_trigger_release.assert_called_with("fake-app", "fake-token", MOCK_RELEASE)
-        mock_wait_for_release.assert_called_with("fake-app", "fake-token", MOCK_RELEASE.version)
-
-    @patch("deploy_heroku.get_latest_heroku_release", return_value=MOCK_RELEASE)
-    @patch("deploy_heroku.get_response", return_value={"commit": MOCK_COMMIT_HASH})
-    @patch("deploy_heroku.do_deploy")
-    def test_main_rollback(self, mock_do_deploy, mock_get_response, mock_get_release):
-        main(MOCK_HEROKU_APP, MOCK_HEROKU_API_TOKEN, MOCK_COMMIT_HASH, json.dumps({"ghd": {"type": "rollback"}}))
-        mock_get_release.assert_called_with("fake-app", "fake-token")
-        mock_get_response.assert_called_with("fake-app", "fake-token", "slugs/227c9e81-a699-9a1bea3b8119")
-        mock_do_deploy.assert_called_once_with("fake-app", "fake-token", "abc1234", rollback=True)
-
-    @patch("deploy_heroku.get_latest_heroku_release", return_value=MOCK_RELEASE)
-    @patch("deploy_heroku.get_response", return_value={"commit": MOCK_COMMIT_HASH})
-    @patch("deploy_heroku.do_deploy")
-    def test_main_deploy(self, mock_do_deploy, mock_get_response, mock_get_release):
-        main(MOCK_HEROKU_APP, MOCK_HEROKU_API_TOKEN, MOCK_COMMIT_HASH, json.dumps({"ghd": {"type": ""}}))
-        mock_get_release.assert_called_with("fake-app", "fake-token")
-        mock_get_response.assert_called_with("fake-app", "fake-token", "slugs/227c9e81-a699-9a1bea3b8119")
-        mock_do_deploy.assert_called_once_with("fake-app", "fake-token", "abc1234", rollback=False)
+    def test_get_deployment_type(self):
+        def test(payload: dict):
+            with tempfile.NamedTemporaryFile(mode="w") as tmpfile:
+                tmpfile.write(json.dumps(payload, indent=2))
+                tmpfile.flush()
+                return get_deployment_type(Path(tmpfile.name))
+
+        self.assertEqual(test({"deployment_type": "rollback"}), "rollback")
```

Comment:
```suggestion
        self.assertEqual(test({"deployment_type": "rollback"}), DeploymentType.ROLLBACK)
```

I would prefer using class `DeploymentType` instead of plain strings everywhere, even in the tests.